### PR TITLE
Model Architecture Summary 🏗️ 

### LINKED ISSUE #282: [FEATURE REQUEST] Output the summary of the model architecture for completeness

### LINKED ISSUE #296: [BUG] Model architecture summary doesn't save up GNN layers by name

### PR DESCRIPTION
**Pull request by KristinaUlicna**
## PR contribution summary

**Why is this PR useful / good for?** *Please describe the problem(s) you're trying to address.*
* Ensures correct & more detailed model architecture & setup arguments archiving

### List of proposed changes / linked issues & discussions

- [x] ✅ Resolves #282
- [x] ✅ Resolves #296

### What should a reviewer concentrate their feedback on?

+ 💻 Code quality
+ 📝 Everything looks OK?

#### What type of PR is this? (check all applicable)

+ 🐛 Bug fix
+ #️⃣ Documentation / code annotation

#### Added tests?

+ 🙅 no, because they aren't needed

---

## PR review summary

Describe what this PR does & **how you reviewed** the individual items, where needed:

* <TEXT>

*Some helper checks to tick off:*

* Focus on image annotation
* Focus on model training
* Could any optimization be applied?
* Is there any redundant code?
* Are there any spelling errors?

**In conclusion**, after my review, I'd like to:

* 🙋 ask some clarifying questions
* 🙅 suggest some specific changes

### CODE REVIEW

**Comment by crangelsmith** [comment]
Changed base to see exact changes wrt previous PR, will change back after reviewing.

File: grace/models/classifier.py [hunk 1]
```diff
@@ -69,18 +67,24 @@ def __init__(
...
         self.classifier_type = classifier_type
+        self.input_channels = input_channels
+        self.hidden_graph_channels = hidden_graph_channels
+        self.hidden_dense_channels = hidden_dense_channels
+        self.node_output_classes = node_output_classes
+        self.edge_output_classes = edge_output_classes
+        self.num_heads = num_heads
+        self.dropout = dropout
+
         hidden_channels_list = [input_channels]
```
Comment (crangelsmith) [inline grace/models/classifier.py:83]: isn't this a duplication of line 75? Do we need the two separate variables (hidden_channels_list and input_channels)?

**Review by crangelsmith — APPROVED** [review]
LGTM and runs well.

File: grace/models/classifier.py (hunk 1 above)
Comment (KristinaUlicna) [reply grace/models/classifier.py:83]: I suppose this assures that the `input_channels` (and `int` == num_features) is put into the list which will grow if other hidden layers are specified. Could be redesigned, agreement on that 😄

**Pull request merged**